CMakeLists.txt

```diff
@@ -71,6 +71,9 @@ if(${CMAKE_SYSTEM_PROCESSOR} MATCHES "armv7l")
   set(CMAKE_C_LINK_FLAGS "${CMAKE_C_LINK_FLAGS} -latomic")
   set(CMAKE_CXX_LINK_FLAGS "${CMAKE_CXX_LINK_FLAGS} -latomic")
 endif()
+if (CMAKE_SYSTEM_PROCESSOR MATCHES "^(arm64|aarch64)")
+  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsigned-char")
+endif()
```

Comment:
I would recommend adding the flag for any architecture, not only for arm64/aarch64. `-fsigned-char` is the default behavior on x86, so if at any point in the future someone wants to build with `-funsigned-char`, it will break the same way on both platforms. I don't know what the default is on other architectures, but since the desired behavior requires the use of `signed char`, it's better to explicitly specify for all platforms.

Reply:
I tested this flag in riscv64. And I found that this could break the linkage of glibc. So, we need to enable _limited_ platforms. I’m against for enabling it all platforms.

Reply:
Or, just providing an option to enable signed char also would be better to implement?

Reply:
Has anyone tried to figure out which parts of the code are actually sensitive to this and simply make them work regardless of char signed-ness? (or use a different type that is well-defined)